### PR TITLE
Fix error handler restoration

### PR DESCRIPTION
**Pull request by eclipxe13**
When call `XsltProcessor::transformToXML()` an internal error handler is set but it is not restored.

This could be because the possible missunderstand that the previous error handler can be set using `\set_error_handler()` but the correct function is `\restore_error_handler()`.

On this PR there is a first commit that probes this failure and a second commit to use the correct functions.

I have a library that depends on genkgo/xsl, I try to upgrade from version `^0.6` to `^1.0` but I can't with this issue.

### CODE REVIEW

**Comment by eclipxe13** [comment]
Please, do not merge yet, I had found the same issue in other 2 places.

**Comment by eclipxe13** [comment]
The process of catching the `LibXmlError` and convert it to a `TransformException` is repeated 3 times in the class, I just refactor it into a single private function. The current test is enough to probe that this is working.

@frederikbosch would you have the chance to review it?

File: src/XsltProcessor.php
Comment (frederikbosch) [inline src/XsltProcessor.php]: Could you merge this line into one

**Review by frederikbosch — CHANGES_REQUESTED** [review]
I like the PR, will merge it after that minor thing is resolved. As a side note: I am not particular happy with error handling in general at the moment. It needs rigorous improvement. When an XSL error is generated, it is not really clear where - file/line - it was generated. I will see if I can improve this in the near future.

File: src/XsltProcessor.php
Comment (eclipxe13) [reply src/XsltProcessor.php]: Would you like me to try other approach? Some ideas:

- Instead of `Closure`, call a private method (i. e. `transformToXmlUnsafe`) and it would show the error location, `transformToXml` would be more like a envelop before calling the private method.

- Break `catchLibXmlErrorThrowTransformException` into `startCatchingLibXmlError` and `stopCatchingLibXmlError`.

- Revert de commit that creates `catchLibXmlErrorThrowTransformException` and just replace `\set_error_handler` with `\restore_error_handler` as needed.

**Comment by eclipxe13** [comment]
@frederikbosch hi, I think I made the changes when you requested.
Would you please take a look? I would like to address this issue. Thanks!

**Review by frederikbosch — APPROVED** [review]

**Comment by frederikbosch** [comment]
Wonderful, looks good.

**Pull request merged**
